Docs: add icon explanations

siddharthisrani opened the pull request:
## :bookmark_tabs: Summary

Added explanations for icons used in the  documentation.

Resolves #5346

## :straight_ruler: Design Decisions

- Included a note explaining the meaning of the 🔥 and -beta icons.

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://mermaid.js.org/community/contributing.html)
- [x] :computer: have added necessary unit/e2e tests.
- [x] :notebook: have added documentation. Make sure [`MERMAID_RELEASE_VERSION`](https://mermaid.js.org/community/contributing.html#update-documentation) is used for all new features.
- [x] :bookmark: targeted `develop` branch

siddharthisrani committed:
Update docs: add icon and info box explanations

siddharthisrani committed:
Added icon explanations in documentation

sidharthv96 reviewed (changes requested):
Can the note be moved to a more general page, so that we don't have to duplicate it inside each diagram pages?

siddharthisrani commented:
Thank you for the feedback, @sidharthv96 .
I agree that having a central location for the icon explanations would be more efficient. I will move the note to a more general page and link to it from the individual diagram pages. I'll update the PR accordingly.

siddharthisrani committed:
Included common markdown content dynamically using script setup

siddharthisrani committed:
Merge branch 'develop' into docs/5346_add-icon-explanations

siddharthisrani commented:
Hi @sidharthv96 ,

I have updated the PR.

Description
1. Implemented a method to dynamically include common markdown content across multiple files using VitePress's <script setup> feature.
2. This approach enhances maintainability and ensures consistency across our documentation.

Please review the changes and let me know if there are any further adjustments needed.

sidharthv96 commented:
> Can the note be moved to a more general page, so that we don't have to duplicate it inside each diagram pages?

I meant having the note in a general page, like the main readme, so that we don't have to duplicate the notes in every page. 

If we do need to duplicate (which I don't think we should), we should use something like vitepress's footer, so that we can inject it in one place, rather than importing in every diagram. 

Also, because we don't use only vitest, including should be done using our custom tag `<!--@include: ./examples.md -->`.

siddharthisrani commented:
Hi @sidharthv96,

I've explored two approaches to include the notes without duplicating them in every diagram page:

Linking to a General Notes Page:

1. Created icons-note.md in the syntax folder.
2 .Updated individual markdown files to refer to the general notes page.

Example code and screenshot below:
For an explanation of the icons used in this documentation, please refer to the [General Icon Explanations]`(./icons-note.md).`

![mermaid2display(1)](https://github.com/user-attachments/assets/b8f4a4b3-457d-4685-b338-3cfe7653ed29)
![mermaid2display](https://github.com/user-attachments/assets/9b047a7e-1e66-4668-a45c-595aa3be61ca)


Using the custom include method:

1. Created icons-note.md in the syntax folder.
2. Used the custom include tag in the individual markdown files.  `<!--@include: ./icons-note.md-->`

Example code and screenshot below:
![mermaid1display](https://github.com/user-attachments/assets/d66b666e-d306-491f-8d52-252d885b7778)

Please let me know which approach you prefer or if there are any other adjustments needed.

sidharthv96 commented:
What I don't understand is why we need the note under every diagram. 
What if we add a `title` to the 🔥 icon, so that it'll show "Newly Added" when user hovers over it?

And whether we should change the icon to something else, so that it's obvious without an explanation.
